doc/vrrp/VRRP_Adaptation_HLD.md

```diff
@@ -0,0 +1,739 @@
+![VRRP ADD/DEL VRRP instance Flows](images/VRRP_Config_Instance_Flow.png)
+
+vrrpmgrd:
+  - Listens to VRRP create, delete and parameter change in CONFIG DB
+  - Upon change
+    - Add/del VRRP instance corresponding Macvlan device to kernel with IPs and state.
+    - Updates VRRP instance configuration to the APPL DB, such as Macvlan device name and Vip.
+    - Update changes to vrrpd by using vtysh commands
+
+vrrpd:
+  - Listens to VRRP create, delete and parameter change in vrrpcfgd
+  - Run VRRP state machine
+    - For VRRP instances state transition to Master, set the Macvlan device into protodown off
+    - For VRRP instances state transition to non-Master, set the Macvlan interface into protodown on
+
+vrrpsyncd:
+  - Listens to Macvlan interface programming in kernel. 
+  - Update the kernel Macvlan device's state to the INTF_table entry of APPL DB.
+
+intforch:
+  - Listens to INTF_Table in APP_DB and updates Vip and virtual MAC entries in ASIC_DB for VRRP instances
+
+##### Uplink interface tracking
+
+![VRRP Tracking interface Flows](images/VRRP_Tracking_Interface_Flow.png)
+
+zebra:
+  - Listen for interfaces state change events notified by the kernel.
+  - Notify interface state change events to vrrpd.
+
+vrrpd:
+  - Match VRRP instance tracking interface and recalculate priority
+    - If track interface goes down, the priority of VRRP instance will be reduced by weight
+    - If track interface goes up, the priority of VRRP instance will be added by weight
+  - Update VRRP packet priority field
+  - Send VRRP packet to Linux Network stack
+
+### SAI API 
+
+VRRP virtual router interface can be created by setting the attribute “SAI_ROUTER_INTERFACE_ATTR_IS_VIRTUAL” with create_router_interface API.
+```
+ * @brief RIF creation is a virtual RIF.
+ *
+ * Create a Virtual RIF object, which only programs the ingress router MAC.
+ * This simplifies the management of VRRP master router's configuration in
+ * SAI adapter, as defined by RFC 5798 (or similar proprietary protocols).
+ * Using a Virtual RIF allows SAI to optimize resources, so neighbor entries
+ * cannot be learned on a Virtual RIF. On a virtual RIF following attributes
+ * are invalid: ADMIN state, MTU size, packet action and multicast enable.
+ * Alternatively VRRP can also be configured using native RIF objects without
+ * using VIRTUAL attribute, with the expectation that SAI adapter will consume
+ * resources that will not be used.
+ *
+ * @type bool
+ * @flags CREATE_ONLY
+ * @default false
+ */
+
+SAI_ROUTER_INTERFACE_ATTR_IS_VIRTUAL,
+
+```
+Considering that some particular vendor SAI implementations may not support this attribute. Use sai_query_attribute_capability to judge if ASIC platform support SAI_ROUTER_INTERFACE_ATTR_IS_VIRTUAL attribute. if not support, the attribute will not added when createvirtual interface with sai interface.
+
+### Configuration and management 
+
+#### Manifest
+
+N/A
+
+#### CLI/YANG Model Enhancements
+
+##### SONiC VRRP YANG MODEL
+
```

Comment:
Please use below file for VRRP sonic yang

https://github.com/dks19/SONiC/blob/master/doc/sonic-vrrp.yang

Reply:
OK ,thanks